policy/file/source.go

```diff
@@ -116,7 +116,7 @@ func (s *Source) MonitorPolicy(ctx context.Context, req policy.MonitorPolicyReq)
 	// handler which starts the evaluation ticker.
 	val, ok := s.policyMap[req.ID]
 	if !ok {
-		req.ErrCh <- fmt.Errorf("failed to get policy %s", req.ID)
+		policy.HandleSourceError(s.Name(), fmt.Errorf("failed to get policy %s", req.ID), req.ErrCh)
```

Comment:
is `string` the best we can do in `HandleSourceError`? is it possible/worthwhile passing `req.ID` in separately for indexing by the metrics destination?

Reply:
At the moment yes because of the way `ErrCh` handles messages. We can look into modifying the behaviour so that the channel accepts a custom wrapped error but this is well out of scope for this change.

Reply:
It is possible to modify the `HandleSourceError` to accept the ID and message and then form the error message before sending it to the channel. This allows us to add the policyID as a label where available on the metric request. This allows individual errors to be associated to policies which is nice.

There are two different behaviours that use this metric point; errors during listing which don't have policyIDs to associate to the metric, and then errors during policy reading which do. Maybe we need different metrics for each failure area to provide more granular information and easier to understand.

Reply:
Yeah, that's what I was thinking. Then I read this:
https://prometheus.io/docs/practices/naming/#labels

And now I think I'm changing my mind. I'm not convinced that it's useful in Prometheus to have the counter for "get policy 7 from `nomad`". It should be a temporary situation where we are continuing to load a policy that doesn't exist on the list; and we are logging the detailed error message.

Let's leave this as is for now; we can add finer-grained monitoring if we discover a need.